bitcoin/src/blockdata/witness.rs

```diff
@@ -14,35 +14,37 @@ use crate::io::{self, Read, Write};
 use crate::prelude::*;
 use crate::VarInt;
 
-/// The Witness is the data used to unlock bitcoins since the [segwit upgrade](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki)
+/// The Witness is the data used to unlock bitcoin since the [segwit upgrade].
 ///
 /// Can be logically seen as an array of byte-arrays `Vec<Vec<u8>>` and indeed you can convert from
 /// it [`Witness::from_vec`] and convert into it [`Witness::to_vec`].
 ///
 /// For serialization and deserialization performance it is stored internally as a single `Vec`,
 /// saving some allocations.
 ///
+/// [segwit upgrade]: <https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki>
 #[derive(Clone, Default, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
 pub struct Witness {
-    /// contains the witness Vec<Vec<u8>> serialization without the initial varint indicating the
-    /// number of elements (which is stored in `witness_elements`)
+    /// Contains the witness Vec<Vec<u8>> serialization without the initial varint indicating the
+    /// number of elements (which is stored in `witness_elements`).
     content: Vec<u8>,
 
-    /// Number of elements in the witness.
-    /// It is stored separately (instead of as VarInt in the initial part of content) so that method
-    /// like [`Witness::push`] doesn't have case requiring to shift the entire array
+    /// The number of elements in the witness.
+    ///
+    /// Stored separately (instead of as a VarInt in the initial part of content) so that methods
+    /// like [`Witness::push`] don't have to shift the entire array.
     witness_elements: usize,
 
-    /// If `witness_elements > 0` it's a valid index pointing to the last witness element in `content`
-    /// (Including the varint specifying the length of the element)
+    /// If `witness_elements > 0` this is a valid index pointing to the last witness element in
+    /// `content` (including the varint specifying the length of the element).
     last: usize,
 
-    /// If `witness_elements > 1` it's a valid index pointing to the second-to-last witness element in `content`
-    /// (Including the varint specifying the length of the element)
+    /// If `witness_elements > 1` this is a valid index pointing to the second-to-last witness
+    /// element in `content` (including the varint specifying the length of the element).
     second_to_last: usize,
 }
 
-/// Support structure to allow efficient and convenient iteration over the Witness elements
+/// Support structure to allow efficient and convenient iteration over the Witness elements.
```

Comment:
This one is pretty unusual - users should already know what an iterator is. "An iterator returning individual witness elements." would be much more informative.